### DISCUSSION TITLE
List of Tutorials :100:

### DISCUSSION THREAD

**Discussion by fcakyon** [obss/sahi#623] [Announcements]
- [Introduction to SAHI](https://medium.com/codable/sahi-a-vision-library-for-performing-sliced-inference-on-large-images-small-objects-c8b086af3b80)

- [Official paper](https://arxiv.org/abs/2202.06934) (NEW)

- [Video inference support is live](https://github.com/obss/sahi/issues/457) (NEW)

- [Kaggle notebook](https://www.kaggle.com/remekkinas/sahi-slicing-aided-hyper-inference-yv5-and-yx)

- [Satellite object detection](https://blog.ml6.eu/how-to-detect-small-objects-in-very-large-images-70234bab0f98)

- [Error analysis plots & evaluation](https://github.com/obss/sahi/issues/356)

- [Interactive result visualization and inspection](https://github.com/obss/sahi/issues/357)

- [COCO dataset conversion](https://medium.com/codable/convert-any-dataset-to-coco-object-detection-format-with-sahi-95349e1fe2b7)

- [Slicing operation notebook](https://github.com/obss/sahi/blob/main/demo/slicing.ipynb)

- `YOLOX` + `SAHI` demo: <a href="https://huggingface.co/spaces/fcakyon/sahi-yolox"><img src="https://raw.githubusercontent.com/obss/sahi/main/resources/hf_spaces_badge.svg" alt="sahi-yolox"></a> (RECOMMENDED)

- `YOLOv5` + `SAHI` walkthrough: <a href="https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_yolov5.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="sahi-yolov5"></a>

- `MMDetection` + `SAHI` walkthrough: <a href="https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_mmdetection.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="sahi-mmdetection"></a>

- `Detectron2` + `SAHI` walkthrough: <a href="https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_detectron2.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="sahi-detectron2"></a>

- [Detectron2+SAHI tutorial](https://medium.com/p/37063e33073f)

Turkish/Turkce:

- [(Github) Yolov5+Sahi Kullanarak Yüksek Doğruluklu Nesne Tespit Uygulamasını Yap](https://github.com/kadirnar/yolov5-pytorch-sahi)

- [(Youtube) Yolov5 ile Küçük Objeleri Daha İyi Tespit Etmek | SAHI & Yolov5 kullanımı](https://www.youtube.com/watch?v=4FmBwH799cs) (YENI)

Spanish:

- [(Youtube) SAHI+DETIC: Detectá Objetos a Gran Escala en Minutos](https://www.youtube.com/watch?time_continue=1&v=r-QGxuMMWfA&feature=emb_title)

**Comment by kadirnar** [comment]
Slicing operation notebook linki 404 vermektedir.

- https://github.com/obss/sahi/blob/main/demo/slicing.ipynb 

Yanllış linklendirme yapılmış galiba

**Comment by fcakyon** [comment]
@kadirnar its not giving any errors for me, can you try from another browser?

<img width="1392" alt="Screen Shot 2022-01-13 at 18 18 57" src="https://user-images.githubusercontent.com/34196005/149357642-a68b2d8e-b01d-47f9-a1e6-431a797835a1.png">

**Comment by kadirnar** [comment]
Firefox ve Chrome üzerinde denedim. Not found 404 hatası veriyor. Bu gayet normal. Çünkü link issues kısmına verilmiş.

```
https://github.com/obss/sahi/issues/demo/slicing.ipynb
```
Link de issues kısmı olmamalı bence. Sizde nasıl açılıyor ona anlam veremedim :/

**Comment by fcakyon** [comment]
@kadirnar thanks a lot mentioning, its fixed now! 🚀

**Comment by rkinas** [comment]
Hi, 
I implemented SAHI and Yolov5 on Kaggle - please look here for more detail: https://www.kaggle.com/remekkinas/sahi-slicing-aided-hyper-inference-yv5-and-yx This is probably first such implementation on Kaggle. Now looking for ways to improve score.

**Comment by fcakyon** [comment]
> Hi, I implemented SAHI and Yolov5 on Kaggle - please look here for more detail: https://www.kaggle.com/remekkinas/sahi-slicing-aided-hyper-inference-yv5-and-yx This is probably first such implementation on Kaggle. Now looking for ways to improve score.

Just looked over your notebook. WOW, its so well prepared 💯 I would gladly help you on your experiments. You can also write to me over [Twitter](https://twitter.com/fcakyon) or [Linkedin](https://www.linkedin.com/in/fcakyon/) for faster response 👍 

Also added your kaggle notebook url to the tutorials list!

**Comment by LawVP-SP** [comment]
Good afternoon. First of all congratulations on the success of your project. I had a question and I don't know if this is the right place. I have been trying to replicate the experiments regarding the dataset named Visdrone but I am having some problems. The parameters to perform the training are as follows:

SGD optimizer:
-Learning rate of 0.01
-Momentum of 0.9
-Weight decay of 0.0001
-Linear warmup of 500 iterations is used. 

How many epochs would you have to set to achieve the results presented in the paper?

On the other hand, when obtaining these results, which classes have been taken into account, all 90 COCO classes, or has it been limited to a series of them?

Thank you in advance

**Comment by fcakyon** [comment]
> Good afternoon. First of all congratulations on the success of your project. I had a question and I don't know if this is the right place. I have been trying to replicate the experiments regarding the dataset named Visdrone but I am having some problems. The parameters to perform the training are as follows:
> 
> SGD optimizer: -Learning rate of 0.01 -Momentum of 0.9 -Weight decay of 0.0001 -Linear warmup of 500 iterations is used.
> 
> How many epochs would you have to set to achieve the results presented in the paper?
> 
> On the other hand, when obtaining these results, which classes have been taken into account, all 90 COCO classes, or has it been limited to a series of them?
> 
> Thank you in advance



> LawVP-SP

Lets continue this discussion in https://github.com/fcakyon/sahi-benchmark/issues/1

**Comment by EKivutha** [comment]
Hey any idea on how to genereate a video from the video, the only feed I get is one with overdrawn bounding boxes

**Comment by JosephXay** [comment]
Could you provide a tutorial to fine-tune models with your proposal in my custom dataset? I really know how to do that

**Comment by fcakyon** [comment]
> Hey any idea on how to genereate a video from the video, the only feed I get is one with overdrawn bounding boxes

Currently there is no video input support, you can extract the frames, perform `sahi` inference on the folder, then convert the frames back into a video.

We are open to any feature pull requests by the way 🚀

**Comment by fcakyon** [comment]
> Could you provide a tutorial to fine-tune models with your proposal in my custom dataset? I really know how to do that

You can find the mmdet config files used for slicing aided fine-tuning on the `xview` and `visdrone` datasets here: https://github.com/fcakyon/sahi-benchmark/tree/main/mmdet_configs

Also check this tutorial for a detailed blog post on slicing aided fine-tuning and inference on `satellite imagery`: https://blog.ml6.eu/how-to-detect-small-objects-in-very-large-images-70234bab0f98

**Comment by JosephXay** [comment]
Thankss ;)

**Comment by JosephXay** [comment]
Hello again! I tested these 2 notebooks:

https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_yolov5.ipynb
https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_detectron2.ipynb

I want to perform slicing aided fine-tuning in a custom dataset. I saw the link that you provide me (https://blog.ml6.eu/how-to-detect-small-objects-in-very-large-images-70234bab0f98) but i dont know how to do the fine-tuning step. Could you update the two notebooks mentioned to perform fine tuning or indicate step by step how to do that?

**Reply by yoloyash** [reply]
> Hello again! I tested these 2 notebooks:
> 
> https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_yolov5.ipynb https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_detectron2.ipynb
> 
> I want to perform slicing aided fine-tuning in a custom dataset. I saw the link that you provide me (https://blog.ml6.eu/how-to-detect-small-objects-in-very-large-images-70234bab0f98) but i dont know how to do the fine-tuning step. Could you update the two notebooks mentioned to perform fine tuning or indicate step by step how to do that?

Hello, did you figure out how to do the fine-tuning? I'm trying to train a yolov8 on satellite imagery where the objects are very small. Any help would be appreciated. Thank you.

**Reply by fcakyon** [reply]
@yoloyash for yolov8, you have to slice the coco formatted dataset using this: https://github.com/obss/sahi/blob/main/docs/slicing.md

Then you train your yolov8 model with slices.

Then you perform sliced inference with sahi+yolov8 as in https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_yolov5.ipynb

**Reply by yoloyash** [reply]
> @yoloyash for yolov8, you have to slice the coco formatted dataset using this: https://github.com/obss/sahi/blob/main/docs/slicing.md
> 
> Then you train your yolov8 model with slices.
> 
> Then you perform sliced inference with sahi+yolov8 as in https://colab.research.google.com/github/obss/sahi/blob/main/demo/inference_for_yolov5.ipynb

Would you recommend I first train it directly on the huge images and then finetune on the slices as mentioned in the paper? Or will directly training on the slices will do the job? Also, which architecture would you recommend be best suitable for smaller object detection? Thanks a ton for the response! I really appreciate it!

**Reply by fcakyon** [reply]
Slicing aided finetuning works best when you include both full and sliced images during training (instead of 2-stage training). So you could definitely try that.

**Reply by yoloyash** [reply]
Okay. That answers my question. Can't thank you enough for you help. :)
